kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/CertificateGenerator.java

```diff
@@ -100,7 +100,10 @@ public static X509Certificate generateSelfSignedX509Certificate(KeyPair pair) {
             var validFrom = Date.from(now);
             var validTo = Date.from(now.plus(Duration.ofDays(9999)));
             var certBuilder = new X509v3CertificateBuilder(
-                    new X500Name("CN=issuer"),
+                    // currently it's important that issuer == subject because we use these certs on vault to authenticate the client
+                    // with client authentication enabled vault sends the subject over as part of the handshake and it needs to match
+                    // an issuer principal in the client keystore, so that it can select the correct key.
+                    new X500Name("CN=localhost"),
```

Comment:
this suggests we should also look at setting up a more realistic test where the client has a certificate signed by an issuer and vault has a client CA store containing the issuer's certificate.

Reply:
Yes, I agree.  I think we need (probably the testing extension) needs something capable of build chains of test certificates.   Maybe there's a test dependency out there we can use.